include/sys/ring_buffer.h

```diff
@@ -21,6 +21,14 @@ extern "C" {
 
 #define SIZE32_OF(x) (sizeof((x))/sizeof(uint32_t))
 
+/* Max size is used internally in algorithm. Value is decreased in the test to
+ * trigger rewind algorithm.
+ */
+#define RING_BUFFER_MAX_SIZE \
+	COND_CODE_1(CONFIG_RING_BUFFER_TEST, (0x00000200), (0x80000000))
```

Comment:
Note that MAX_SIZE is used as a threshold to trigger "rewind" algorithm. Algorithm is used to prevent 32 bit wrapping and by default it is set to huge value (2^31). In the ring buffer test it is decreased to ensure that it is triggered and tested. Threshold is reduced to 512 since the biggest ring buffer used in the test has 256 bytes. Asserts are added to detect case when test uses bigger buffers (it is highly unlikely that application will ever exceed the 2Gb limit),